src/main/java/edu/ksu/canvas/attendance/entity/AttendanceAssignment.java

```diff
@@ -0,0 +1,158 @@
+package edu.ksu.canvas.attendance.entity;
+
+
+import javax.persistence.*;
+import java.io.Serializable;
+
+@Entity
+@Table(name = "attendance_assignment")
+public class AttendanceAssignment implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "assignment_id")
+    private Long assignmentId;
+
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "section_id", foreignKey = @ForeignKey(name = "fk_section_attendance"), nullable = false)
+    private AttendanceSection attendanceSection;
+
+    @Column(name = "canvas_assignment_id")
+    private Long canvasAssignmentId;
+
+    @Column(name = "assignment_points")
+    private Double assignmentPoints;
+
+    @Column(name = "present_points")
+    private Double presentPoints;
+
+    @Column(name = "tardy_points")
+    private Double tardyPoints;
+
+    @Column(name = "excused_points")
+    private Double excusedPoints;
+
+    @Column(name = "absent_points")
+    private Double absentPoints;
+
+    @Column(name = "grading_on", nullable = false)
+    private boolean gradingOn = false;
+
+    @Column(name = "assignment_name")
+    private String assignmentName;
+
+    @Transient
+    public Status status;
+
+    public enum Status{
+        UNKNOWN,
+        OKAY,
+        NOT_LINKED_TO_CANVAS,
+        CANVAS_AND_DB_DISCREPANCY
+    }
+
+    public AttendanceAssignment() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1186)